src/plugins/data/public/search/expressions/esaggs.ts

```diff
@@ -56,9 +56,11 @@ export function getFunctionDefinition({
         searchSource,
       } = await getStartDependencies();
 
-      const aggConfigsState = JSON.parse(args.aggConfigs);
-      const indexPattern = await indexPatterns.get(args.index);
-      const aggConfigs = aggs.createAggConfigs(indexPattern, aggConfigsState);
+      const indexPattern = await indexPatterns.create(args.index.value, true);
```

Comment:
I used `indexPatterns.create` here, because manually creating `new IndexPattern()` still requires you to provide the field formats service, plus some UI settings... which means adding more dependencies to `esaggs` (currently ui settings is not required).

However we could still refactor to use `new IndexPattern()` directly if desired; this was just a simpler change.